### PR TITLE
Add vertical ruler for writer

### PR DESCRIPTION
**Pull request by Darshan-upadhyay1110**
- add new LOK callback for vertical & horizontal ruler update

Change-Id: I3c0e26f22072de4612e128d58ac41b629be82807

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by Darshan-upadhyay1110** [comment]
CORE Patch: https://gerrit.libreoffice.org/c/core/+/167481

**Review by gokaysatir — CHANGES_REQUESTED** [review]
Okay, for vertical ruler: We only have header and footer margins that are draggable as far as i can see.

So i guess we shouldn't have code for break points inside VRuler file?

And also it seems to me that vertical ruler goes off screen as i play with the header footer settings :) It feels like something related to left / right margins are forgotten. This is a guess. Can you check that vertical ruler doesn't go off screen and is readable.

I'd check the code from scratch and make some polishing (if required).

Thanks @Darshan-upadhyay1110 :) We are so close.

**Comment by Darshan-upadhyay1110** [comment]
> So i guess we shouldn't have code for break points inside VRuler file?

Yes we do not need break point for vertical ruler

> And also it seems to me that vertical ruler goes off screen as i play with the header footer settings :) It feels like something related to left / right margins are forgotten. This is a guess. Can you check that vertical ruler doesn't go off screen and is readable.

Can you please give me hint, how did you change the header footer settings ? What action you performed that made ruler goes out of screen ?

**Comment by gokaysatir** [comment]
> Can you please give me hint, how did you change the header footer settings ? What action you performed that made ruler goes out of screen ?

Of course, i used page settings and set the header & footer margins.

File: browser/src/control/VRuler.js [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+/*
+ * Ruler Handler
+ */
+
+/* global L _ */
+L.Control.VRuler = L.Control.extend({
```
Comment (eszkadev) [inline browser/src/control/VRuler.js:16]: In the new code please avoid using Leaflet's code (L.Control / Class etc...).
We should use native `class` and also TypeScript.
Then we will be able to convert horizontal ruler and make some base class for both to share code.

File: browser/src/control/VRuler.ts [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+			tabs: [],
+			unit: null,
+			DraggableConvertRatio: null,
+			timer: null,
+			showruler: true,
+			position:'topleft',
+			disableMarker: false
+		};
+
+		constructor(map: ReturnType<typeof L.map>, options: Options) {
```
Comment (eszkadev) [inline browser/src/control/VRuler.ts:87]: did you run: `make prettier-write` to improve formatting?

File: browser/.beforeprettier [hunk 3]
```diff
@@ -75,6 +75,7 @@
 /src/control/Parts.js
 /src/control/Permission.js
 /src/control/Ruler.js
+/src/control/VRuler.ts
```
Comment (eszkadev) [inline browser/.beforeprettier:78]: please not add new files to ignore list, we want new formatting to be applied

File: browser/src/control/VRuler.ts (hunk 2 above)
Comment (Darshan-upadhyay1110) [reply browser/src/control/VRuler.ts:87]: no, i did not run that. i just do select all then format code by right click in VS Code

File: browser/.beforeprettier (hunk 3 above)
Comment (Darshan-upadhyay1110) [reply browser/.beforeprettier:78]: okay

Comment (Darshan-upadhyay1110) [reply browser/.beforeprettier:78]: i see now code is formatted, I will push that change

**Review by eszkadev — APPROVED** [review]
Looks good, lets get this in and then do a followup :)

**Comment by eszkadev** [comment]
One UX comment:
- dragging blue arrow works
- dragging grey line shows tooltip but doesn't move margin (does selection instead)

I would expect to have both of them draggable and working.

**Comment by eszkadev** [comment]
@gokaysatir you requested changes, please re-check

**Review by gokaysatir — APPROVED** [review]

**Pull request merged**